021_MergeTwoSortedLists.py

```diff
@@ -0,0 +1,84 @@
+class ListNode(object):
+     def __init__(self, x):
+         self.val = x
+         self.next = None
+class ListNode(object):
+     def __init__(self, x):
+         self.val = x
+         self.next = None
+
+class Solution(object):
+    def mergeTwoLists(self, l1, l2):
+        last=None
+        lr=None
+        while (l1!=None) and (l2!=None):
+            if l1.val<=l2.val:
+                if lr==None:
+                    lr=l1
+                    last=l1
+                else: 
+                    last.next=l1
+                    last=last.next
+                l1=l1.next
+                
+            else:
+                if (l2.val<l1.val):
+                    if lr==None:
+                        lr=l2
+                        last=lr
+                    else: 
+                        last.next=l2 
+                        last=last.next
+                l2=l2.next
+                       
+        if (l1==None):
+            last.next=l2
+        else:
+            if (l2==None):
+                last.next=l1
+        return lr
+            
+def arrayToList(arr):
+    l=None
+    for i in range(len(arr)):
+        if l==None:
+            l=ListNode(arr[i])
+            last=l
+        else:
+            pom=ListNode(arr[i])
+            last.next=pom
+            last=last.next
+    return l
+def listToArray(l):
```

Comment:
It makes more sense for this to be function in ListNode, as you are actually here working on instance "l" of ListNode

Reply:
I'm not sure I did this right way.

Reply:
I corrected it.